src/codec.rs

```diff
@@ -223,6 +223,13 @@ pub trait EncodeAppend {
 	fn append(self_encoded: Vec<u8>, to_append: &[Self::Item]) -> Result<Vec<u8>, Error>;
 }
 
+/// Trait that allows the length of a collection to be read, without having
+/// to read and decode the entire elements.
+pub trait EncodeLength {
```

Comment:
I would like to call it `DecodeLength` or something similar, but the current name implies we want to encode a length.